### ISSUE TITLE
Python built-in map() to support more than one iterable

### ISSUE THREAD

**Issue by Jeroendevr** [TranscryptOrg/Transcrypt#862]
Currently the `map()` [built in function](https://docs.python.org/3.9/library/functions.html#map) is well supported, however it only works when passing one iterable. When passing two iterables the iterable result is `null`

Excpects
```py
words = ("bike", "steer")

def func_1(prin):
    print(prin[0])

def func3(prin):
    print(prin[3])

function = (func_1, func3)

def apply(func: callable, word):
    func(word)

mapped = map(apply, function, words)

for x in mapped:
    x

---
b
e
```
Reality
```py
null
null
```

**Comment by JennaSys** [comment]
The way `map` is implemented right now, it only accepts a single iterable. But I'll take a look at expanding that to match CPython behavior.

**Comment by JennaSys** [comment]
The new implementation of `map` should work correctly once v3.9.3 is deployed (currently it's only on the dev branch). Though, the variable `function` in the above example should be changed to `functions` so as not to confuse the JavaScript side of things.

**Comment by JennaSys** [comment]
Merged to master on PR #876
